odoo/http.py

```diff
@@ -1479,6 +1483,35 @@ def _dispatch_nodb():
         except werkzeug.exceptions.HTTPException as e:
             return e(environ, start_response)
 
+    def get_dispatch_context_managers(self, request):
+        """ Return the context managers to use for dispatching a request.
+        This includes ``request`` itself, and optionally a profiler.
+        """
+        context_managers = [request]
+        if request.session.profile_session and request.session.db:
+            if 'set_profiling' in request.httprequest.path:
```

Comment:
why the special-case, and why handle it globally here? can't `ir.profile.set_profiling()` do the right thing?

Reply:
It was another solution but:
- the profiler is already strated
- the profiler is not exposed anywhere so we cannot access it to stop it easily.
- this is a way to protect the route/rpc so that if anything goes really wrong, we can always disable the profiler.